File: gaia-sdk-javascript/src/graphql/response/type/UpdatedStatementImpulse.ts

import {Statement} from "./Statement";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update statement impulse
*/
export interface UpdatedStatementImpulse {
    id?:Uuid, 
    /**
    * the statement instance
    */
    data?:Statement
}

File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedStatementImpulse.py

from gaia_sdk.graphql.response.type.Statement import Statement

from dataclasses import dataclass
from typing import List
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool
from gaia_sdk.graphql.request.enumeration.RuntimeState import RuntimeState
from gaia_sdk.graphql.request.enumeration.SkillState import SkillState
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField

@dataclass
class UpdatedStatementImpulse:
    """
    Impulse which indicates the result of a update statement impulse
    """
    dictionary: dict
    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the statement instance
    """
    @property
    def data(self) -> Statement:
        return Statement(self.dictionary.get("data"))
